redis/src/cluster_async/mod.rs

```diff
@@ -356,23 +370,34 @@ where
             (target, Err(err)) => {
                 trace!("Request error {}", err);
 
+                let request = this.request.as_mut().unwrap();
+                if request.retry >= this.retry_params.number_of_retries {
+                    self.respond(Err(err));
+                    return Next::Done.into();
+                }
+                request.retry = request.retry.saturating_add(1);
+                let sleep_duration = this.retry_params.wait_time_for_retry(request.retry);
+
                 let address = match target {
-                    OperationTarget::Node { address } => address,
+                    OperationTarget::Node { address } => Some(address),
                     OperationTarget::FanOut => {
                         // Fanout operation are retried per internal request, and don't need additional retries.
                         self.respond(Err(err));
                         return Next::Done.into();
                     }
+                    OperationTarget::NotFound => {
+                        println!("target not found. retry: {}", request.retry);
+                        // TODO - this is essentially a repeat of the retriable error. probably can remove duplication.
+                        let request = this.request.take().unwrap();
+                        return Next::RefreshSlots {
```

Comment:
I've tested this branch against a common failure scenario, complete loss of a node. Based on that testing, we may want to do something like this here:
```
                        let mut request = this.request.take().unwrap();
                        request.info.redirect = None; 
```                        
Else, the request is likely to be repeatedly retried against the failed node, the issue being that when the node initially fails, the client will get an erroneous `MOVED` error pointing to the failed node. If we don't unset the redirect somehow, the request will never connect to the correct node after a slot-refresh has occurred.

Reply:
did you test this change, or is this only an assumption?

where exactly in the code do you think that the redirect should be reset?

Reply:
I tested your branch locally, using same scenarios I'd previously tested [here](https://github.com/redis-rs/redis-rs/pull/964#issuecomment-1736744056), so it's not just an assumption.

I'm not immediately sure the best way to accomplish this, but the code snippet I posted above at line 391 at least causes the request to eventually recover. Without it, it will go through a cycle of 16 retries before eventually failing, long after the cluster has recovered.

Reply:
The more general scenario I suppose, is that a request with a redirect may, in the process of retrying, need to try the original request again without the redirect. Right now the code doesn't allow for this flexibility.

Reply:
Thanks for the data! I'll give it a thought and see how to handle this. Hopefully another patch next week :)